doc/neps/nep-0041-improved-dtype-support.rst

```diff
@@ -0,0 +1,706 @@
+=================================================
+NEP 41 — First step towards a new Datatype System
+=================================================
+
+:title: Improved Datatype Support
+:Author: Sebastian Berg
+:Author: Add...
+:Status: Draft
+:Type: Standard Track
+:Created: 2020-02-03
+
+
+.. note::
+
+    This NEP is part of a series of NEPs encompassing first information
+    about the previous dtype implementation and issues with it in NEP 40.
+    NEP 41 (this document) then provides an overview and generic design
+    choices for the refactor.  
+    Further NEPs 42 and 43 go into the technical dtypes of the datatype
+    and universal function related internal and external API changes.
+    In some cases it may be necessary to consult the other NEPs for a full
+    picture of the desired changes and why these changes are necessary.
+
+
+Abstract
+--------
+
+`Datatypes <data-type-objects-dtype>` in NumPy describe how to interpret each
+element in arrays. NumPy provides ``int``, ``float``, and ``complex`` numerical
+types, as well as string, datetime, and structured (or `~numpy.recarray`)
```

Comment:
I don't think we need to mention recarray here. Can we just link to the docs for structured dtypes, which work just fine without recarray?